test/test_other.py

```diff
@@ -3484,6 +3484,34 @@ def test_jspi_add_function(self):
       '-Wno-experimental']
     self.do_runf('other/test_jspi_add_function.c', 'done')
 
+  @requires_jspi
+  def test_jspi_async_function(self):
+    # Make sure async library functions are not automatically JSPI'd.
```

Comment:
I'm actually a bit confused as to how this test verifies it. Wouldn't it silently pass w/o the change in this PR too?

Reply:
Without the change this test will error with `SuspendError: trying to suspend without WebAssembly.promising`.